src/AuthorizedSocketConnection.js

```diff
@@ -107,14 +114,25 @@ export default class AuthorizedSocketConnection<TContext, TCredentials> {
       variableValues: variables,
       contextValue: {
         ...this.config.context,
-        subscribe: async (
-          channel,
-          parseMessage = this.config.defaultParseMessage,
-        ) => {
-          const source = this.config.redis.subscribe(channel);
-          const parsed = AsyncUtils.map(source, parseMessage);
-          const filtered = AsyncUtils.filter(parsed, this.isAuthorized);
-          return filtered;
+        subscribe: async (...args) => {
+          const source = this.config.subscriber.subscribe(...args);
+          const filtered = AsyncUtils.filter(
+            source.iterable,
+            this.isAuthorized,
+          );
+
+          return {
+            next: () => filtered.next(),
+            throw: err => filtered.throw(err),
+            return: async () => {
+              await source.close();
+              if (filtered.return) await filtered.return();
+            },
+            // $FlowFixMe
+            [Symbol.asyncIterator]() {
+              return this;
+            },
```

Comment:
Note the API change here from subscribe. This is the simplest way I know to ensure that close is called on `return()`, this works though, because i know graphqljs specifically calls the return on the original iterable (this one ^)

Reply:
what if this just returns the `AsyncQueue`, and the consumer just needs to call `queue.iterable` and `queue.close`?

it's not a perfect API, but all this is internal to this module anyway, and it won't make it look like the async iterable supports some property that it in fact does not

Reply:
you can't return the queue right? Eventually you need to hand off the iterable to GQL and that needs to be the one filtered one, not the queue.iterable. I'm not sure either where the consumer would call close? they don't get a call back when they need to tear down. We could pass store the queue with the subscriptions and manually tear it down, but it doesn't let the user adjust the behavio similarly if they need too. IDK i don't love this, but it seems like the most in keeping with the api. and the wrapped `return` thing I think is an intended thing you can do...

Reply:
yeah, so you hand the iterable off to GQL, but in your local map of subscriptions, you store the queue (really the closable?) instead of necessarily the async iterable itself

the `subscriptions` map is here anyway, right? alternatively we could make `Subscriber` return something that has `Symbol.asyncIterator` _and_ has a `close` method

Reply:
yeah i thought of doing that first but the book keeping isn't straightforward, you can have multiple queues per subscription, saving it off the subs map didn't quite work because that's a promise at this point maybe. its wonky. I can give it another shot tho

Reply:
how do you get multiple queues per subscription?

right now we only hang on to these to check (1) what's already subscribed and (2) to close the subscriptions, no?

the API should be the same if you store the queue... you just need to call `.close()` instead of `.return()`

Reply:
> the API should be the same if you store the queue... you just need to call .close() instead of .return()

We'd probably need to call both, the subscription iterator isn't the iterator returned here, it's something subscribe _makes_ and then returns from `subscribe`,

Reply:
ick

Reply:
so we need `return()` to make the async iterable clean up?

Reply:
ugh i am not at all sure anymore. that was my impression but let me mess around a bit more